spec/acceptance/tic-tac-toe_spec.rb

```diff
@@ -108,27 +108,52 @@ def save_result(result)
   end
 
   it 'wont let the same person play twice' do
-    HumanPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).has_turn(0)
-    response = HumanPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).has_turn(1)
+    player_turn(0)
+    response = player_turn(1)
     expect(response).to eq(errors: [:one_turn_per_player])
   end
 
   it 'will make the ai pick the winning move out of two choices' do
-    new_board = board_set_up([:X, 0], [:O, 1], [:X, 5], [:O, 4])
-    AIPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).execute
+    new_board = board_set_up( [:O, 0], [:X, 3], [:O, 1], [:X, 5])
+    ai_turn
     response = view_grid.execute({})
-    expect(response).to eq(grid: [[:X, :O, nil], [nil, :O, :X], [nil, :O, nil]], result: :winner)
+    expect(response).to eq(grid: [[:O, :O, :O], [:X, nil, :X], [nil, nil, nil]], result: :winner)
   end
 
   it 'will make the ai pick the draw when optimal' do
     new_board = board_set_up([:O, 0], [:X, 1], [:O, 2], [:X, 4], [:O, 7], [:X, 3])
-    AIPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).execute
-    HumanPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).has_turn(8)
-    AIPlayer.new(update_grid: UpdateGrid.new(game_state: game_state)).execute
+    ai_turn
+    player_turn(8)
+    ai_turn
     response = response = view_grid.execute({})
     expect(response).to eq(grid: [[:O, :X, :O], [:X, :X, :O], [:O, :O, :X]], result: :draw)
   end
 
+  it 'will make the ai pick a spot the minimizes the human and maximises the computer' do
+    new_board = board_set_up([:X, 0], [:O, 1], [:X, 2], [:O, 3], [:X, 6])
+    ai_turn
+    player_turn(5)
+    ai_turn
+    response = view_grid.execute({})
+    expect(response).to eq(grid: [[:X, :O, :X], [:O, :O, :X], [:X, :O, nil]], result: :winner)
+  end
+
+  it 'will make the ai pick a spot the minimizes the human and maximises the computer with 4 options' do
+    new_board = board_set_up([:X, 0], [:O, 4], [:X, 6], [:O, 3], [:X, 5])
+    ai_turn
+    response = view_grid.execute({})
+    expect(response).to eq(grid: [[:X, :O, nil], [:O, :O, :X], [:X, nil, nil]], result: nil)
+  end
+
+  it 'will make the ai play to a draw against a perfect human' do
+    player_turn(0)
+    ai_turn
+    player_turn(8)
+    ai_turn
+    response = view_grid.execute({})
+    expect(response).to eq(grid: [[:X, :O, nil], [nil, :O, nil], [nil, nil, :X]], result: nil)
+  end
+
   private
```

Comment:
Good call to make them private 👏 

(Otherwise classes and methods defined inside RSpec blocks have a nasty habit of leaking out to the rest of the code.)